Support pino-http 9.x

Thank you for this useful library! Currently using `nestjs-pino@3.5.0` with `pino-http` which now has a new release `9.0.0` available. As far as I can tell, the only breaking changes in that release were changes to the supported node versions. Please add that version as a supported `peerDependency` so we can stay on the latest `pino-http`

```
❯ npm i --no-progress
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: <snip>
npm WARN Found: pino-http@8.6.1
npm WARN node_modules/pino-http
npm WARN   pino-http@"9.0.0" from the root project
npm WARN   1 more (nestjs-pino)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer pino-http@"^6.4.0 || ^7.0.0 || ^8.0.0" from nestjs-pino@3.5.0
npm WARN node_modules/nestjs-pino
npm WARN   nestjs-pino@"3.5.0" from the root project
```

--- a/.github/workflows/build-lint-test.yml
+++ b/.github/workflows/build-lint-test.yml
@@ -13,7 +13,6 @@ jobs:
           - "9"
           - "10"
         nodejs-version:
-          - 16
           - 18
           - 20
         include:

--- a/package.json
+++ b/package.json
@@ -60,7 +60,7 @@
     "jest": "29.7.0",
     "memorystream": "^0.3.1",
     "pino": "^8.15.0",
-    "pino-http": "^8.4.0",
+    "pino-http": "^9.0.0",
     "prettier": "^3.0.2",
     "reflect-metadata": "^0.1.13",
     "rimraf": "^5.0.1",
@@ -73,6 +73,6 @@
   },
   "peerDependencies": {
     "@nestjs/common": "^8.0.0 || ^9.0.0 || ^10.0.0",
-    "pino-http": "^6.4.0 || ^7.0.0 || ^8.0.0"
+    "pino-http": "^6.4.0 || ^7.0.0 || ^8.0.0 || ^9.0.0"
   }
 }

